python/plugins/processing/gui/BatchAlgorithmDialog.py

```diff
@@ -57,9 +57,16 @@ def __init__(self, alg, parent=None):
     def runAsSingle(self):
         self.close()
 
-        from processing.gui.AlgorithmDialog import AlgorithmDialog
+        dlg = (
+            self.algorithm()
+            .create()
+            .createCustomParametersWidget(parent=iface.mainWindow())
+        )
+        if not dlg:
+            from processing.gui.AlgorithmDialog import AlgorithmDialog
+
+            dlg = AlgorithmDialog(self.algorithm().create(), parent=iface.mainWindow())
```

Comment:
it's possible the python garbage collector will clean up the created algorithm too early here:

```suggestion
        alg_instance = self.algorithm().create()
        dlg = (
            alg_instance.createCustomParametersWidget(parent=iface.mainWindow())
        )
        if not dlg:
            from processing.gui.AlgorithmDialog import AlgorithmDialog

            dlg = AlgorithmDialog(alg_instance, parent=iface.mainWindow())
```